### DISCUSSION TITLE
Image cache for thumbnails when browsing for songs and any other online images

### DISCUSSION THREAD

**Discussion by AeroBliss** [ppy/osu#22943] [Ideas]
This will help with devices that have slow internet connections since they will not have to download the thumbnail again (usually experienced on slow mobile connections). It also makes for a nicer browsing experience since you will not see gray boxes when browsing through the same songs again.
The cache could be wiped on session closure or every day/week/month, and a setting could be added to wipe image cache.

**Comment by peppy** [comment]
I was pretty sure this was tracked somewhere in an issue already but can't seem to find it.

This is definitely something we want to do (not just for thumbnails, but user avatars and basically any other online images). It should be added to `OnlineStore` or similar.

**Reply by Joehuu** [reply]
The issue was made by a user that deleted their account. I was able to get the remnants from Discord:
![image](https://user-images.githubusercontent.com/35318437/227407825-08188080-b520-4b51-9082-a11ec7d1a91c.png)

**Reply by peppy** [reply]
So bad...

**Reply by AeroBliss** [reply]
Updated title to address everything else image related that needs to be downloaded

**Comment by Ladwick** [comment]
This sounds great, I definitely would love getting this feature in the game since my internet is pretty slow!
